Foundation/NSDecimal.swift

```diff
@@ -56,74 +187,286 @@ extension Decimal {
+    }
+    public init(_ value: Int64) {
+        self.init(Double(value))
+    }
+    public init(_ value: UInt) {
+        self.init(UInt64(value))
+    }
+    public init(_ value: Int) {
+        self.init(Int64(value))
+    }
+    public var isSignalingNaN: Bool {
+        return false
+    }
+    public static var nan: Decimal {
+        return quietNaN
+    }
+    public static var quietNaN: Decimal {
+        var quiet = Decimal()
+        quiet._isNegative = 1
+        return quiet
+    }
+    public var floatingPointClass: FloatingPointClassification {
+        if _length == 0 && _isNegative == 1 {
+            return .quietNaN
+        } else if _length == 0 {
+            return .positiveZero
+        }
+        if _isNegative == 1 {
+            return .negativeNormal
+        } else {
+            return .positiveNormal
+        }
+    }
+    public var isSignMinus: Bool {
+        return _isNegative != 0
+    }
+    public var isNormal: Bool {
+        return !isZero && !isInfinite && !isNaN
+    }
+    public var isFinite: Bool {
+        return !isNaN
+    }
+    public var isZero: Bool {
+        return _length == 0 && _isNegative == 0
+    }
+    public var isSubnormal: Bool {
+        return false
+    }
+    public var isInfinite: Bool {
+        return false
+    }
+    public var isNaN: Bool {
+        return _length == 0 && _isNegative == 1
+    }
+    public var isSignaling: Bool {
+        return false
+    }
 }
 
 extension Decimal : CustomStringConvertible {
     public init?(string: String, locale: Locale? = nil) {
-        // Avoid a compiler warning for now by not calling NSUnimplemented
-        return nil
+        let scan = Scanner(string: string)
+        var theDecimal = Decimal()
+        if !scan.scanDecimal(&theDecimal) {
+            return nil
+        }
+        self = theDecimal
+    }
+    public var description: String {
+        var val = self
+        return NSDecimalString(&val, nil)
     }
-    public var description: String { NSUnimplemented() }
 }
 
-public func NSDecimalIsNotANumber(_ dcm: UnsafePointer<Decimal>) -> Bool { NSUnimplemented() }
-
+public func NSDecimalIsNotANumber(_ dcm: UnsafePointer<Decimal>) -> Bool {
+	return dcm.pointee.isNaN
+}
 
 /***************	Operations		***********/
-public func NSDecimalCopy(_ destination: UnsafeMutablePointer<Decimal>, _ source: UnsafePointer<Decimal>) { NSUnimplemented() }
+public func NSDecimalCopy(_ destination: UnsafeMutablePointer<Decimal>, _ source: UnsafePointer<Decimal>) {
+    destination.pointee.__lengthAndFlags = source.pointee.__lengthAndFlags
+    destination.pointee.__exponent = source.pointee.__exponent
+    destination.pointee.__reserved = source.pointee.__reserved
+    destination.pointee._mantissa = source.pointee._mantissa
+}
 
-public func NSDecimalCompact(_ number: UnsafeMutablePointer<Decimal>) { NSUnimplemented() }
+public func NSDecimalCompact(_ number: UnsafeMutablePointer<Decimal>) {
+    number.pointee.compact()
```

Comment:
We can't mark them as deprecated and still use them to power the extended methods on NSDecimal (perhaps we could rename them so that they are not publicly autocompleted; but that is not going to happen likely for Swift 3.X).

Often when making the overlay I had to use a temp copy to transact upon instead. And we can't make them hoisted since they transact upon an address of a Decimal not the Decimal itself.